### PR TITLE
212 password change

### LINKED ISSUE #212: Settings: change password feature
Current process:
`Logout  --> Forgot my password --> Password nova`

### PR DESCRIPTION
**Pull request by Lobosque**
Fixes #212 

* Updates `UpdateUserInfoForm` to accept password change;
* Add "Change Password" section in the settings forms;
* Modify the settings form javascript to work for N tabs

### CODE REVIEW

**Comment by Lobosque** [comment]
@dethos How do you think the new password form should behave for users that registered using GithubOauth? (there is no current password for these cases).

File: humans/templates/humans/update_user_form.html [hunk 1]
```diff
@@ -65,6 +66,19 @@
                     </div>
                   </li>
                   <li class="section" id="section3">
+                    <div class="col-xxs-12 col-xs-12 col-md-12 start-xs">
+                      <div class="form__block">
+                        {{ form.current_password.errors }} {{ form.current_password.label_tag }} {{ form.current_password }}
+                      </div>
+                      <div class="form__block">
+                        {{ form.new_password2.errors }} {{ form.new_password1.label_tag }} {{ form.new_password1 }}
```
Comment (nunosilva800) [inline humans/templates/humans/update_user_form.html:76]: I think this is a typo, should be `{{ form.new_password1.errors }}`

**Comment by Lobosque** [comment]
I did it on purpose to always display the same message to both of them!

**Comment by dethos** [comment]
@Lobosque for users who signed up using Github, we should not display the change password tab or to make it clear, instead of the form a message could be displayed saying that since they used Oauth there is not need to update passwords.

By the way, thanks for your contribution. As soon as you hide the tab or display that message, I will make a full review and merge the PR (on a first look everything appears to be fine :+1: )

File: humans/static/javascripts/update_user_form.js [hunk 2]
```diff
@@ -5,33 +5,16 @@ $(document).ready(function(){
...
-  });
-  $("#tab3").click(function(){
-    //slide up all the link lists
-    $("#section3").show();
-    $("#section1").hide();
-    $("#section2").hide();
-    $("#tab3").addClass("active");
-    $("#tab1").removeClass("active");
-    $("#tab2").removeClass("active");
+  $(".sett__nav__item").click(function(evt) {
```
Comment (dethos) [inline humans/static/javascripts/update_user_form.js:8]: For some reason this event is not being added to `#tab4` (at least on Firefox)

**Review by dethos — CHANGES_REQUESTED** [review]
After testing a bit I found a small issue. 

Also `#tab4` is missing here: https://github.com/whitesmith/hawkpost/pull/254/files#diff-634d4c9620895e0cdf312c07c3ad90b6R49

File: humans/static/javascripts/update_user_form.js (hunk 2 above)
Comment (Lobosque) [reply humans/static/javascripts/update_user_form.js:8]: I tested here on Firefox and it triggers the event just fine.
Maybe a version issue? Can you try again?

**Review by Lobosque — COMMENTED** [review]
@dethos I added a commit to hide the change password section from github signups.
However I couldn't reproduce the tab problem. All of them worked just fine in both Chrome and Firefox.

File: humans/forms.py [hunk 3]
```diff
@@ -21,16 +22,61 @@ class Meta:
...
+    new_password2 = forms.CharField(label=_('New password confirmation'),
+                                    required=False,
+                                    widget=forms.PasswordInput)
+
     def __init__(self, *args, **kwargs):
+        # Flag to let the save method know when to call set_password
+        self.change_password = False
         self.pub_key = None
-        return super().__init__(*args, **kwargs)
+        return super(UpdateUserInfoForm, self).__init__(*args, **kwargs)
```
Comment (dethos) [inline humans/forms.py]: Why this change? This is a Python 3 project, there is no need to call super using the old "style".

File: humans/forms.py [hunk 4]
```diff
@@ -21,16 +22,61 @@ class Meta:
...
+
+    def save(self, commit=True):
+        new_password = self.cleaned_data.get('new_password2')
+        if self.change_password:
+            self.instance.set_password(new_password)
+        return super(UpdateUserInfoForm, self).save(commit=commit)
```
Comment (dethos) [inline humans/forms.py]: here super can also be changed to the new style

**Review by dethos — COMMENTED** [review]
Just a small observation

File: humans/static/javascripts/update_user_form.js (hunk 2 above)
Comment (dethos) [reply humans/static/javascripts/update_user_form.js:8]: Just tested again, with different browsers and it is working fine. Not sure what was the problem the other time.

**Review by dethos — APPROVED** [review]

**Pull request merged**